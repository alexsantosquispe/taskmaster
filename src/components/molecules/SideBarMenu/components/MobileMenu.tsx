import { CloseIcon, MenuIcon } from '../../../../icons';

import { twMerge } from 'tailwind-merge';
import { ToggleButton } from '../../../atoms/ToggleButton/ToggleButton';
import { LogoHeader } from './SideBarHeader';

interface MobileMenuProps {
  isOpen: boolean;
  setIsOpen: (value: boolean) => void;
}

export const MobileMenu = ({ isOpen, setIsOpen }: MobileMenuProps) => {
  return (
    <div className={twMerge('flex items-center justify-between p-2 md:hidden')}>
      <LogoHeader isCollapsed={false} />
      <ToggleButton
        isCollapsed={isOpen}
        setIsCollapsed={setIsOpen}
        expandIcon={<CloseIcon />}
        collapseIcon={<MenuIcon />}
      />
    </div>
  );
};
