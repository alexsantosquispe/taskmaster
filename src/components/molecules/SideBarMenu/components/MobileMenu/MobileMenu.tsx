import { CloseIcon, MenuIcon } from '@/icons';

import { LogoHeader } from '../SideBarHeader';
import { ToggleButton } from '@/components/atoms/ToggleButton/ToggleButton';
import { twMerge } from 'tailwind-merge';

interface MobileMenuProps {
  isOpen: boolean;
  setIsOpen: (value: boolean) => void;
  className?: string;
}

export const MobileMenu = ({
  isOpen,
  setIsOpen,
  className
}: MobileMenuProps) => {
  return (
    <div
      data-testid="mobile-menu"
      className={twMerge(
        'flex h-[3.5rem] items-center justify-between pr-2 pl-4 md:hidden',
        className
      )}
    >
      <LogoHeader isCollapsed={false} />
      <ToggleButton
        isCollapsed={isOpen}
        setIsCollapsed={setIsOpen}
        expandIcon={<CloseIcon />}
        collapseIcon={<MenuIcon />}
      />
    </div>
  );
};
