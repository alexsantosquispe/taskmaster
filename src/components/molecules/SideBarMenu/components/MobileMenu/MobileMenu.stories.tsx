import type { Meta, StoryObj } from '@storybook/react';

import { useState } from 'react';
import { MobileMenu } from './MobileMenu';

const meta: Meta<typeof MobileMenu> = {
  title: 'Components/Molecules/SideBarMenu/MobileMenu',
  component: MobileMenu,
  parameters: {
    layout: 'centered'
  }
};

export default meta;

type Story = StoryObj<typeof MobileMenu>;

export const Default: Story = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <MobileMenu
      isOpen={isOpen}
      setIsOpen={setIsOpen}
      className="w-[30rem] md:flex"
    />
  );
};

Default.args = {};
