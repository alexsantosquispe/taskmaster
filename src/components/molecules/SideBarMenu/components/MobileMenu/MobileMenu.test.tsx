import { render } from '@testing-library/react';
import { MobileMenu } from './MobileMenu';

describe('MobileMenu', () => {
  const props = {
    isOpen: false,
    setIsOpen: jest.fn()
  };

  it('should render the component correctly', () => {
    const component = render(<MobileMenu {...props} />);

    expect(component).toMatchSnapshot();
  });
});
