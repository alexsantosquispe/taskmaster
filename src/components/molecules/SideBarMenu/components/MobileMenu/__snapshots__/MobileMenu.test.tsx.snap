// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`MobileMenu should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-[3.5rem] items-center justify-between pr-2 pl-4 md:hidden"
        data-testid="mobile-menu"
      >
        <div
          class="flex items-center gap-x-2"
        >
          <svg
            class="size-6 rounded bg-blue-700 p-[0.1875rem] text-white dark:bg-orange-600"
            fill="none"
            height="16"
            id="Box--Streamline-Majesticons"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <desc>
              
    Box Streamline Icon: https://streamlinehq.com
  
            </desc>
            <path
              clip-rule="evenodd"
              d="M7.6466666666666665 1.4346666666666668a0.6666666666666666 0.6666666666666666 0 0 1 0.7066666666666667 0l4.602666666666666 2.876666666666667L8 7.226666666666667 3.0439999999999996 4.311333333333333l4.602666666666666 -2.876666666666667zM2.005333333333333 5.247333333333334A0.6673333333333332 0.6673333333333332 0 0 0 2 5.333333333333333v5.333333333333333a0.6666666666666666 0.6666666666666666 0 0 0 0.3133333333333333 0.5653333333333332L7.333333333333333 14.369333333333332v-5.9879999999999995L2.005333333333333 5.246666666666666zM8.666666666666666 14.369333333333332l5.02 -3.1373333333333333A0.6666666666666666 0.6666666666666666 0 0 0 14 10.666666666666666V5.333333333333333c0 -0.028666666666666663 -0.002 -0.057999999999999996 -0.005333333333333333 -0.086L8.666666666666666 8.380666666666666v5.988666666666667z"
              fill="currentColor"
              fill-rule="evenodd"
              stroke-width="0.6667"
            />
          </svg>
          <h1
            class="text-lg md:block"
          >
            <span
              class="font-bold"
            >
              Task
            </span>
            Master
          </h1>
        </div>
        <button
          aria-label="Collapse menu"
          class="rounded-lg p-2 text-neutral-600 hover:cursor-pointer hover:bg-neutral-100 hover:text-blue-700 dark:text-white/60 dark:hover:bg-neutral-800 dark:hover:text-orange-500"
        >
          <svg
            class="size-5"
            data-testid="menu-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 12h18"
            />
            <path
              d="M3 18h18"
            />
            <path
              d="M3 6h18"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-[3.5rem] items-center justify-between pr-2 pl-4 md:hidden"
      data-testid="mobile-menu"
    >
      <div
        class="flex items-center gap-x-2"
      >
        <svg
          class="size-6 rounded bg-blue-700 p-[0.1875rem] text-white dark:bg-orange-600"
          fill="none"
          height="16"
          id="Box--Streamline-Majesticons"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <desc>
            
    Box Streamline Icon: https://streamlinehq.com
  
          </desc>
          <path
            clip-rule="evenodd"
            d="M7.6466666666666665 1.4346666666666668a0.6666666666666666 0.6666666666666666 0 0 1 0.7066666666666667 0l4.602666666666666 2.876666666666667L8 7.226666666666667 3.0439999999999996 4.311333333333333l4.602666666666666 -2.876666666666667zM2.005333333333333 5.247333333333334A0.6673333333333332 0.6673333333333332 0 0 0 2 5.333333333333333v5.333333333333333a0.6666666666666666 0.6666666666666666 0 0 0 0.3133333333333333 0.5653333333333332L7.333333333333333 14.369333333333332v-5.9879999999999995L2.005333333333333 5.246666666666666zM8.666666666666666 14.369333333333332l5.02 -3.1373333333333333A0.6666666666666666 0.6666666666666666 0 0 0 14 10.666666666666666V5.333333333333333c0 -0.028666666666666663 -0.002 -0.057999999999999996 -0.005333333333333333 -0.086L8.666666666666666 8.380666666666666v5.988666666666667z"
            fill="currentColor"
            fill-rule="evenodd"
            stroke-width="0.6667"
          />
        </svg>
        <h1
          class="text-lg md:block"
        >
          <span
            class="font-bold"
          >
            Task
          </span>
          Master
        </h1>
      </div>
      <button
        aria-label="Collapse menu"
        class="rounded-lg p-2 text-neutral-600 hover:cursor-pointer hover:bg-neutral-100 hover:text-blue-700 dark:text-white/60 dark:hover:bg-neutral-800 dark:hover:text-orange-500"
      >
        <svg
          class="size-5"
          data-testid="menu-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 12h18"
          />
          <path
            d="M3 18h18"
          />
          <path
            d="M3 6h18"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
